authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerHelper.java

```diff
@@ -445,7 +445,10 @@ protected RangerPolicy createPolicyAddResources(RangerMetadataObject metadataObj
     List<String> nsMetadataObject = metadataObject.names();
     for (int i = 0; i < nsMetadataObject.size(); i++) {
       RangerPolicy.RangerPolicyResource policyResource =
-          new RangerPolicy.RangerPolicyResource(nsMetadataObject.get(i));
+          new RangerPolicy.RangerPolicyResource(
+              nsMetadataObject.get(i),
+              false,
+              metadataObject.type().equals(RangerMetadataObject.Type.PATH));
```

Comment:
I think we need to abstract a function in the `RangerAuthorizationPlugin` class.
```
protected abstract RangerPolicy createPolicyAddResources(AuthorizationMetadataObject metadataObject)
```

because `RangerAuthorizationHDFSPlugin` and `RangerAuthorizationHadoopSQLPlugin` have different Ranger policy resource expression.
- RangerAuthorizationHadoopSQLPlugin is `db.tab.column`
- RangerAuthorizationHDFSPlugin is `path` and other params.

@theoryxu What do you think?

Reply:
OK, I have refactored this part.